src/db/transaction.cr

```diff
@@ -0,0 +1,118 @@
+module DB
+  abstract class Transaction
+    include Disposable
+    include BeginTransaction
+
+    abstract def connection : Connection
+
+    def commit
+      close!
+    end
+
+    def rollback
+      close!
+    end
+
+    private def close!
+      raise DB::Error.new("Transaction already closed") if closed?
```

Comment:
I wasn't thinking about _querying_ the status, but rather simply issuing the `close` statement to the DB, and catch the "already closed" exception. However, either option is ok for me.